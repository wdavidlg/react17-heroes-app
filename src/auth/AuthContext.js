import { createContext, useEffect, useReducer } from "react";
import { authReducer } from "./authReducer";


export const AuthContext = createContext(null);

const initialState = {
    logged: false
}

const init = (initialState) => {
    return JSON.parse(localStorage.getItem('user')) || initialState;
}


const AuthProvider = ({children}) => {

    const [user, dispatch] = useReducer(authReducer, initialState, init)

    useEffect(() => {
        localStorage.setItem('user', JSON.stringify(user));
    }, [user])

    return (
        <AuthContext.Provider value={{
            user, 
            dispatch
        }}>
            {children}
        </AuthContext.Provider>
    )
}

export default AuthProvider;