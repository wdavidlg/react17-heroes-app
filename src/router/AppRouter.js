import { BrowserRouter as Router, Switch } from "react-router-dom";
import LoginScreen from "../componentes/login/LoginScreen";
import React, { useContext } from 'react'
import DashboardRoutes from "./DashboardRoutes";
import PrivateRoute from "./PrivateRoute";
import { AuthContext } from "../auth/AuthContext";
import PublicRoute from "./PublicRoute";

const AppRouter = () => {

    const {user} = useContext(AuthContext);
    const {logged} = user;

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute 
                        isAuthenticate={logged}
                        exact 
                        path="/login" 
                        component={LoginScreen} />
                    <PrivateRoute
                        path='/'
                        isAuthenticate = {logged}
                        component={DashboardRoutes} />
                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter

