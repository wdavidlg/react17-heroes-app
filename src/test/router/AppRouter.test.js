import {mount} from 'enzyme'
import { AuthContext } from '../../auth/AuthContext'
import AppRouter from '../../routers/AppRouter'

describe('Pruebas en <AppRouter />', () => {

    const user = {
        logged: false
    }
    
    test('debe de mostrar el login si no esta autenticado', () => {
        const user = {
            logged: false
        }
        const wrapper = mount(
            <AuthContext.Provider value = {{
                user
            }}>
                <AppRouter/>
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar el componente marvel si esta autenticado', () => {
        const user = {
            name: 'will',
            logged: true
        }
        const wrapper = mount(
            <AuthContext.Provider value = {{
                user
            }}>
                <AppRouter/>
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text()).toBe('Marvel Comics')
    })
    
    
})
