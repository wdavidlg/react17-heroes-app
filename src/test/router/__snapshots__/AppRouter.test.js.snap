// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el componente marvel si esta autenticado 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Asociaciones
        </a>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-bs-target="#navbarNavAltMarkup"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav me-auto mb-2 mb-lg-0"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
            <a
              aria-current={null}
              className="nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
            <button
              className="nav-link btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </div>
          <span
            className="navbar-text text-info"
          >
            will
          </span>
        </div>
      </div>
    </nav>,
    <div
      className="container mt-5"
    >
      <div>
        <h1>
          Marvel Comics
        </h1>
        <hr />
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-spider"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Spider Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-spider"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-captain"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Captain America
                    </h5>
                    <p
                      className="card-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-captain"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-iron"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Iron Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-iron"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-thor"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Thor
                    </h5>
                    <p
                      className="card-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-thor"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-hulk"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hulk
                    </h5>
                    <p
                      className="card-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-hulk"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-wolverine"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Wolverine
                    </h5>
                    <p
                      className="card-text"
                    >
                      James Howlett
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-wolverine"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-daredevil"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Daredevil
                    </h5>
                    <p
                      className="card-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-daredevil"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-hawkeye"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hawkeye
                    </h5>
                    <p
                      className="card-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-hawkeye"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-cyclops"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Cyclops
                    </h5>
                    <p
                      className="card-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-cyclops"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card mb-3 animate__animated animate__heartBeat"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <img
                    alt="marvel-silver"
                    className="img-fluid rounded-start"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-sm-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Silver Surfer
                    </h5>
                    <p
                      className="card-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      href="/heroe/marvel-silver"
                      onClick={[Function]}
                    >
                      Ver m치s...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
